build.gradle

```diff
@@ -62,3 +67,85 @@ processResources {
         exclude 'mcmod.info'
     }
 }
+
+// because the normal output has been made to be obfuscated
+task deobfJar(type: Jar) {
+    from sourceSets.main.output
+    classifier = 'dev'
+}
+
+// add a source jar
+task sourceJar(type: Jar, dependsOn: "sourceMainJava") {
+    from "build/sources/java"
+    classifier = 'sources'
+}
+
+// add a javadoc jar
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from 'build/docs/javadoc'
+}
+
+// deployement stuff
+
+configurations { deployerJars }
+
+dependencies { deployerJars "org.apache.maven.wagon:wagon-ssh:2.2" }
+
+artifacts {
+    archives sourceJar
+    archives deobfJar
+    archives javadocJar
+}
+
+
+uploadArchives {
+    repositories {
+        mavenDeployer {
+            configuration = configurations.deployerJars
+
+            repository(url: "http://files.minecraftforge.net/maven/manage/upload") {
+                authentication(userName: "AbrarSyed", password: project.hasProperty(forgeMavenPass) ? null : project.getProperty('forgeMavenPass')) // the elvis operator. look it up.
```

Comment:
@AbrarSyed I havnt set up an account or anything for this, and ill probally need some help setting all this up. I have made APIs like this in the past, but using my website as a maven. I dont currently have anything for an API to be used for but Im sure itll come it usfull later.

Reply:
The password for that is injected at build-time by the forge jenkins. Once you get a set of your own credentials, we can change it.

Upon deployment, it magically shows up here: http://files.minecraftforge.net/SecretRoomsMod/
About that.. Im worried the group needs to be the same for it to work correctly.. il double check with Overmind on that... In maven the group decides the file structure, and that script is generated by crawling the file structure... so.. uh.. yeah..

Reply:
So will Overmind set me up with an account or is that something I will have to do. Ive used maven before but I used it through GitHub and GitHub pages

Reply:
Dunno yet, Il find out. For now just leave my credentials in there.

Reply:
ok